src/functions/govNotify.ts

```diff
@@ -30,45 +32,40 @@ const govNotify: Handler = async (event: SQSEvent, context?: Context, callback?:
   const notifyConfig = await Configuration.getInstance().getNotifyConfig();
   const notifyClient = new NotifyClient(notifyConfig.api_key);
   const notifyService: NotificationService = new NotificationService(notifyClient);
-  const notifyPromises: Array<Promise<any>> = [];
+  const batchItemFailures: SQSBatchItemFailure[] = [];
 
-  event.Records.forEach((sqsRecord: SQSRecord) => {
-    const objectPutEvent: S3Event = JSON.parse(sqsRecord.body);
+  for (const sqsRecord of event.Records) {
+    try {
+      const objectPutEvent: S3Event = JSON.parse(sqsRecord.body);
 
-    if (objectPutEvent.Records) {
-      objectPutEvent.Records.forEach((s3Record: S3EventRecord) => {
-        const s3Object: any = s3Record.s3.object;
-        // Object key may have spaces or unicode non-ASCII characters.
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-        const decodedS3Key = decodeURIComponent(s3Object.key.replace(/\+/g, ' '));
+      if (objectPutEvent.Records) {
+        for (const s3Record of objectPutEvent.Records) {
+          const s3Object: any = s3Record.s3.object;
+          // Object key may have spaces or unicode non-ASCII characters.
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+          const decodedS3Key = decodeURIComponent(s3Object.key.replace(/\+/g, ' '));
+          const notifyPartialParams = await downloadService.getCertificate(decodedS3Key);
 
-        if (decodedS3Key.includes('VOSA')) {
-          const emailList = process.env.TFL_EMAIL_LIST?.split(',');
-          emailList?.forEach((email) => {
-            const notifyPromise = downloadService.getCertificate(decodedS3Key).then((notifyPartialParams: IPartialParams) => {
-              if (!notifyPartialParams.shouldEmail || notifyPartialParams.shouldEmail === 'true') {
+          if (!notifyPartialParams.shouldEmail || notifyPartialParams.shouldEmail === 'true') {
+            if (decodedS3Key.includes('VOSA')) {
+              const emailList = process.env.TFL_EMAIL_LIST?.split(',');
+              for (const email of emailList ?? []) {
```

Comment:
be inclined to set that as an empty string, since this is a string in the variable which is email addresses, using commas as the delimiter
```suggestion
              const emailList = process.env.TFL_EMAIL_LIST?.split(',') ?? '';
              for (const email of emailList) {
```

Reply:
Not sure i would want to null coalesce that to a string would I? split returns an array, and I need it to be an array below. I can see the value in moving the coalesce up a line tho

Reply:
Oh yes sorry my bad, yes - can the array be moved up a line please